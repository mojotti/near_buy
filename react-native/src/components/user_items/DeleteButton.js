import React, { Component } from 'react';
import { Alert, Text, TouchableHighlight } from 'react-native';
import PropTypes from 'prop-types';

import { styles } from '../../static/styles/UserItemDetailsStyles';
import { DELETION_ERROR, localhost } from '../../static/constants';
import { getBearerHeaders } from '../../networking/networking';

export default class DeleteButton extends Component {
  _navigateBack = () => {
    const { goBack } = this.props.navigation;
    goBack();
  };

  _removeItemAndNavigateBack = () => {
    const url = `http://${localhost}:5000/api/v1.0/user/items/${this.props.id}`;
    fetch(url, {
      method: 'DELETE',
      headers: getBearerHeaders(this.props.token),
    })
      .then(response => response.json())
      .then(responseJson => {
        if (responseJson.ok) {
          this.props.fetchItems();
          this._navigateBack();
        }
      })
      .catch(error => {
        Alert.alert(...DELETION_ERROR);
        console.error(error);
      });
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this._removeItemAndNavigateBack}
        style={styles.deleteButton}
      >
        <Text style={styles.buttonText}>Delete item</Text>
      </TouchableHighlight>
    );
  }
}

DeleteButton.propTypes = {
  id: PropTypes.number.isRequired,
  token: PropTypes.string.isRequired,
};
