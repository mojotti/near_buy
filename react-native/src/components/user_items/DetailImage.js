import React, { Component } from 'react';
import { Dimensions, View } from 'react-native';
import ImageLoad from 'react-native-image-placeholder';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
const { height } = Dimensions.get('window');

export default class DetailImage extends Component {
  renderImage = () => {
    return (
      <View style={{ flex: 1 }}>
        <ImageLoad
          style={{ flex: 1 }}
          placeholderStyle={{
            flex: 1,
            alignSelf: 'center',
            alignItems: 'center',
            justifyContent: 'center',
          }}
          source={{ uri: this.props.url }}
          isShowActivity={false}
        />
      </View>
    );
  };

  renderPlaceholder = () => {
    return (
      <View
        style={{
          flex: 1,
          alignSelf: 'center',
          alignItems: 'center',
          justifyContent: 'center',
          paddingBottom: 40,
          backgroundColor: '#FFFFFF',
        }}
      >
        <MaterialIcons name="add-a-photo" size={height * 0.4} color="gray" />
      </View>
    );
  };

  render() {
    return this.props.url ? this.renderImage() : this.renderPlaceholder();
  }
}
