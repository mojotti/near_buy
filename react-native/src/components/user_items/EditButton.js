import React, { Component } from 'react';
import { Alert, Text, TouchableHighlight } from 'react-native';
import PropTypes from 'prop-types';

import { styles } from '../../static/styles/UserItemDetailsStyles';
import { DELETION_ERROR, localhost } from '../../static/constants';
import { getHeadersForRegistering } from '../../networking/networking';

export default class EditButton extends Component {
  _getHeaders = () => {
    const headers = new Headers();
    headers.append('Authorization', `Bearer ${this.props.token}`);
    headers.append('Content-Type', 'application/json');
    return headers;
  };

  _navigateBack = () => {
    const { goBack } = this.props.navigation;
    goBack();
  };

  _updateItemAndNavigateBack = () => {
    const url = `http://${localhost}:5000/api/v1.0/items/${this.props.id}`;
    console.log(this.props.location);
    fetch(url, {
      method: 'PUT',
      headers: this._getHeaders(),
      body: JSON.stringify({
        id: this.props.item.id,
        title: this.props.item.title,
        description: this.props.item.description,
        price: this.props.item.price,
        longitude: this.props.location.longitude,
        latitude: this.props.location.latitude,
        sold: false,
      }),
      N,
    })
      .then(response => response.json())
      .then(responseJson => {
        console.log('response', responseJson);
        this.props.fetchItems();
        this._navigateBack();
      })
      .catch(error => {
        Alert.alert(...DELETION_ERROR);
        console.error(error);
      });
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this._updateItemAndNavigateBack}
        style={styles.editButton}
      >
        <Text style={styles.buttonText}>Edit item</Text>
      </TouchableHighlight>
    );
  }
}

EditButton.propTypes = {
  id: PropTypes.number.isRequired,
  token: PropTypes.string.isRequired,
};
