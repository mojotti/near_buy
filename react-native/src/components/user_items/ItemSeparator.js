import React, { Component } from 'react';
import { View } from 'react-native';
import { styles } from '../../static/styles/ListViewItemStyle';

export default class ItemSeparator extends Component {
  render() {
    return <View style={styles.separator} />;
  }
}
