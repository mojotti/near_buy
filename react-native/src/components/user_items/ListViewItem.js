import React from 'react';
import { Image, Text, View } from 'react-native';
import { styles } from '../../static/styles/ListViewItemStyle';
import { localhost } from '../../static/constants';

export class ListViewItem extends React.Component {
  render() {
    const imagePath = `http://${localhost}:5000/${
      this.props.itemId
    }/image0.jpg`;

    return (
      <View style={{ flexDirection: 'row', flex: 1 }}>
        <Image
          style={{ width: 50, height: 50 }}
          source={{
            uri: imagePath,
          }}
        />
        <View style={[styles.item, styles.separator]}>
          <Text>{`Title: ${this.props.item.title}`}</Text>
          <Text>{`Description: ${this.props.item.description}`}</Text>
          <Text>{`Price: ${this.props.item.price} â‚¬`}</Text>
        </View>
      </View>
    );
  }
}
