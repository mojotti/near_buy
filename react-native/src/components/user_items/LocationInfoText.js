import React from 'react';
import { Text, View } from 'react-native';
import { styles } from '../../static/styles/UserItemDetailsStyles';

export default class LocationInfoText extends React.Component {
  render() {
    return (
      <View style={{ alignSelf: 'stretch' }}>
        <Text
          style={styles.infoText}
        >{`\u2022 Your exact location is not visible to others`}</Text>
        <Text style={styles.infoText}>
          {`\u2022`}
          <Text style={{ color: 'blue' }}> Blue</Text>
          <Text>{` color indicates your item's location`}</Text>
        </Text>
        <Text style={styles.infoText}>
          {`\u2022`}
          <Text style={{ color: 'red' }}> Red</Text>
          <Text>{` color indicates your location`}</Text>
        </Text>
      </View>
    );
  }
}
