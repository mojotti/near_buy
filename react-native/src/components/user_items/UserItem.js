import React, { PureComponent } from 'react';
import { Alert, Text, TouchableOpacity, View } from 'react-native';
import { connect } from 'react-redux';
import ImageLoad from 'react-native-image-placeholder';

import Swipeout from 'react-native-swipeout';
import { styles } from '../../static/styles/ListViewItemStyle';
import { DELETION_ERROR, localhost } from '../../static/constants';

export class _UserItem extends PureComponent {
  _getHeaders() {
    const headers = new Headers();
    headers.append('Authorization', `Bearer ${this.props.token}`);
    return headers;
  }

  _onPress = () => {
    const { navigate } = this.props.navigation;
    navigate('UserItemDetails', { item: this.props.item, fetchItems: this.props.fetchItems });
  };

  _removeItem(itemId) {
    const url = `http://${localhost}:5000/api/v1.0/user/items/${itemId}`;
    fetch(url, {
      method: 'DELETE',
      headers: this._getHeaders(),
    })
      .then(response => response.json())
      .then(responseJson => {
        if (responseJson.ok) {
          this.props.fetchItems();
        }
      })
      .catch(error => {
        console.error(error);
        Alert.alert(...DELETION_ERROR);
      });
  }

  render() {
    const deleteButton = [
      {
        text: 'Delete',
        backgroundColor: '#ff1018',
        onPress: () => {
          this._removeItem(this.props.item.id);
        },
      },
    ];

    const imagePath = `http://${localhost}:5000/api/v1.0/${
      this.props.item.id
    }/image0.jpg`;

    return (
      <View>
        <Swipeout right={deleteButton} autoClose={true}>
          <TouchableOpacity onPress={this._onPress}>
            <View style={styles.itemContainer}>
              <ImageLoad
                style={styles.image}
                placeholderStyle={{ height: 70, resizeMode: 'contain' }}
                source={{ uri: imagePath }}
              />
              <View style={styles.item}>
                <Text style={styles.title}>{this.props.item.title}</Text>
                <Text style={styles.price}>{`${this.props.item.price} â‚¬`}</Text>
              </View>
            </View>
          </TouchableOpacity>
        </Swipeout>
      </View>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    username: state.authorizationReducer.username,
    token: state.authorizationReducer.token,
  };
};

const UserItem = connect(mapStateToProps, null)(_UserItem);
export default UserItem;
