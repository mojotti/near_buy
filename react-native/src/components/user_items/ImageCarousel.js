import React from 'react';
import PropTypes from 'prop-types';
import Carousel from 'react-native-looped-carousel';
import { styles } from '../../static/styles/UserItemDetailsStyles';
import DetailImage from './DetailImage';

export default class ImageCarousel extends React.Component {
  render() {
    return (
      <Carousel
        style={styles.carousel}
        autoplay={false}
        pageInfo
        currentPage={0}
        pageInfoBackgroundColor={'#292929'}
        pageInfoTextStyle={{ color: '#FFFFFF' }}
    // arrows are disabled for now since they don't work as intended on ios
        arrows={false}
        arrowStyle={styles.arrowStyle}
        leftArrowText={' '}
        rightArrowText={' '}
      >
        <DetailImage
          url={this.props.imageUrls[0]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[1]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[2]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[3]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
      </Carousel>
    );
  }
}

ImageCarousel.propTypes = {
  imageUrls: PropTypes.arrayOf(PropTypes.string).isRequired,
  id: PropTypes.number.isRequired,
  onImageUpload: PropTypes.func.isRequired,
  numOfPics: PropTypes.number.isRequired,
};
