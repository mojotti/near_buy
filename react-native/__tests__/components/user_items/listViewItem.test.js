import React from 'react';
import renderer from 'react-test-renderer';

import { _UserItem } from '../../../src/components/user_items/UserItem';
import { mockedListViewItem } from '../../../src/static/samples/mockedListViewItem';

describe('_UserItem', () => {
  test('renders and shows current details', () => {
    const rendered = renderer
      .create(<_UserItem item={mockedListViewItem} />)
      .toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
});
