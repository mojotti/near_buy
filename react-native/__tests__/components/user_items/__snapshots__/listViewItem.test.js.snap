// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_UserItem renders and shows current details 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dbddde",
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {},
          Object {
            "left": 0,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "position": "relative",
                },
                Object {
                  "borderRadius": 100,
                  "height": 112.5,
                  "margin": 10,
                  "overflow": "hidden",
                  "width": 112.5,
                },
              ]
            }
          >
            <Image
              borderRadius={undefined}
              onError={[Function]}
              onLoadEnd={[Function]}
              resizeMode={undefined}
              source={
                Object {
                  "uri": "http://localhost:5000/api/v1.0/2/image0.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#e9eef1",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  Object {
                    "borderRadius": undefined,
                  },
                  Object {},
                ]
              }
            >
              <ActivityIndicator
                animating={true}
                color="gray"
                hidesWhenStopped={true}
                size="small"
                style={
                  Object {
                    "margin": "auto",
                    "position": "absolute",
                    "zIndex": 9,
                  }
                }
              />
              <Image
                source={
                  Object {
                    "testUri": "../../react-native-image-placeholder/Images/empty-image.png",
                  }
                }
                style={
                  Object {
                    "height": 70,
                    "resizeMode": "contain",
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flex": 1,
                "flexDirection": "column",
                "padding": 10,
                "width": 750,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
            >
              Item 1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              50 â‚¬
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View />
    <View />
  </View>
</View>
`;
